### DISCUSSION TITLE
Log username in a table on a webpage

### DISCUSSION THREAD

**Discussion by JR-Vermeer** [spatie/laravel-activitylog#898] [Q&A] [answered]
Hello,

I have a table on my webpage where I log the activities from users. But I have a question now. How do I log the name of the person who changed something? I unfortunately couldn't figure it out with the docs.

![image](https://user-images.githubusercontent.com/70522775/120171436-56fbd580-c202-11eb-9b0d-532aca9a228a.png)
![image](https://user-images.githubusercontent.com/70522775/120171518-6a0ea580-c202-11eb-8e31-5a68028ff2f8.png)
![image](https://user-images.githubusercontent.com/70522775/120171571-7abf1b80-c202-11eb-8f17-7bda585134d6.png)

Kind Regards,
Hennie

**Comment by Gummibeer** [comment, marked as answer]
Hey, you can use the `causer` relationship.

```php
$activities = $poster->activities->load('causer');

foreach($activities as $activity) {
    echo $activity->causer->name;
}
```

**Reply by JR-Vermeer** [reply]
Thank you for your quick response! Where should I place this piece of code? In my controller, my blade or my model?

**Reply by Gummibeer** [reply]
The first line is the eager loading of the relationship (preventing `n+1` queries).
https://laravel.com/docs/8.x/eloquent-relationships#eager-loading

The `foreach` loop is plain PHP but could be transformed to blade. Important is the `$activity->causer->xyz`.  causer` is the relationship that resolves to your cause - likely the user model. And `name` or `xyz` is the attribute you want to access on that causer instance.
Keep in mind that the `causer` is optional - in case you are on PHP8 you can solve this by using the PHP8 null safe operator.
https://stitcher.io/blog/php-8-nullsafe-operator

**Reply by JR-Vermeer** [reply]
I'm getting this error now 😬
(Sorry for bothering you, it's for a school project, I'm new to programming)

![image](https://user-images.githubusercontent.com/70522775/120177253-844b8200-c208-11eb-85fe-e1bb2d92bc40.png)

**Reply by Gummibeer** [reply]
In your examples it's likely `$events` - you will have to pass the variable/data to the view to use it.
For general Laravel questions you should use https://larachat.co or https://laracasts.com/discuss or any other community, please.
